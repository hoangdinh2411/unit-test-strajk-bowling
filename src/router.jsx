import { createBrowserRouter } from 'react-router-dom';

import Booking from './views/Booking';
import Confirmation from './views/Confirmation';

const router = createBrowserRouter(
  [
    {
      path: '/',
      element: <Booking />,
    },
    {
      path: '/confirmation',
      element: <Confirmation />,
    },
  ],
  {
    basename: import.meta.env.VITE_BASENAME || '/',
  }
);

export default router;
