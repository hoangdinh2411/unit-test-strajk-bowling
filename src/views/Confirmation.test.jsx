import { beforeEach } from 'vitest';
import Confirmation from './Confirmation';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { RouterProvider } from 'react-router-dom';
import router from '../router';
describe('Testing Confirmation page', () => {
  beforeEach(() => {
    render(
      <RouterProvider router={router}>
        <Confirmation />
      </RouterProvider>
    );
  });
  it('Should show the message if there is no booking', () => {
    fireEvent.click(screen.getByTestId('navigation-icon'));
    fireEvent.click(screen.getByText('Confirmation'));
    expect(screen.getByText('See you soon!')).toBeInTheDocument();
    expect(screen.getByText('Inga bokning gjord!')).toBeInTheDocument();
  });
});
