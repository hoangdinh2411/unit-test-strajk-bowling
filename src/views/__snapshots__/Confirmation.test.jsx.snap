// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing Confirmation page > Should show the message if there is no booking 1`] = `
<h1
  class="top__title"
>
  See you soon!
</h1>
`;

exports[`Testing Confirmation page > Should show the message if there is no booking 2`] = `
<h2
  class="confirmation__no-booking"
>
  Inga bokning gjord!
</h2>
`;
